### PR TITLE
make HoldoffNotifier delay constants configurable

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim kim.andrewsy@gmail.com

Fixes https://github.com/heptio/contour/issues/592 so we can use HoldoffNotifier in e2e tests. 

cc @davecheney

### CODE REVIEW

File: internal/contour/holdoff.go [hunk 1]
```diff
@@ -42,16 +42,36 @@ type HoldoffNotifier struct {
...
+
+func (hn *HoldoffNotifier) SetHoldoffMaxDelay(t time.Duration) {
+	hn.holdoffMaxDelay = t
 }
 
 func (hn *HoldoffNotifier) OnChange(builder *dag.Builder) {
 	hn.mu.Lock()
 	defer hn.mu.Unlock()
+
+	if hn.holdoffDelay == 0 {
```
Comment (andrewsykim) [inline internal/contour/holdoff.go]: This check is only needed once, I was thinking of removing this check by using a constructor func but this seemed cleaner.

Comment (davecheney) [reply internal/contour/holdoff.go]: What happens if someone calls `SetHoldoffDelay(0)`, their value will be overridden with the default.

File: internal/contour/holdoff.go [hunk 2]
```diff
@@ -42,16 +42,36 @@ type HoldoffNotifier struct {
...
+
+	holdoffDelay    time.Duration
+	holdoffMaxDelay time.Duration
+}
+
+func (hn *HoldoffNotifier) SetHoldoffDelay(t time.Duration) {
+	hn.holdoffDelay = t
+}
+
+func (hn *HoldoffNotifier) SetHoldoffMaxDelay(t time.Duration) {
```
Comment (davecheney) [inline internal/contour/holdoff.go]: What happens if these functions are called after the notifier is started? Its either going to be a data race or the updated values will be ignored.

**Comment by andrewsykim** [comment]
Good point @davecheney. How do you feel about having a constructor func for HoldoffNotifier that sets the defaults in that case?

**Comment by davecheney** [comment]
I'd prefer to punt on this issue til its assigned a milestone. There's more than enough to do in 0.11 and 0.12

**Comment by davecheney** [comment]
Closing based on https://github.com/heptio/contour/pull/927#issuecomment-472251124, please resend this when #592 is assigned a milestone.

**Pull request closed without merging**